From: Clement Lefebvre <clement.lefebvre@linuxmint.com>
Date: Thu, 3 May 2018 16:34:25 +0100
Subject: Startup settings: Use icons in button box

In verbose languages, if any of the button text is too long,
they all get wide (consistent sizes) and the box gets bigger
than the window, this results in the buttons disappearing altogether
until the window is resized.
---
 .../share/cinnamon/cinnamon-settings/modules/cs_startup.py   | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/files/usr/share/cinnamon/cinnamon-settings/modules/cs_startup.py b/files/usr/share/cinnamon/cinnamon-settings/modules/cs_startup.py
index a28cfa3..9f254fb 100755
--- a/files/usr/share/cinnamon/cinnamon-settings/modules/cs_startup.py
+++ b/files/usr/share/cinnamon/cinnamon-settings/modules/cs_startup.py
@@ -428,24 +428,28 @@ class AutostartBox(Gtk.Box):
         box.set_halign(Gtk.Align.CENTER)
         button_holder.add(box)
 
-        self.add_button = Gtk.Button.new_with_label(_("Add"))
+        self.add_button = Gtk.Button.new_from_icon_name("list-add-symbolic", Gtk.IconSize.BUTTON)
+        self.add_button.set_tooltip_text(_("Add"))
         self.add_button.connect("clicked", self.on_add_button_clicked)
         button_group.add_widget(self.add_button)
         box.add(self.add_button)
 
-        self.edit_button = Gtk.Button.new_with_label(_("Edit"))
+        self.edit_button = Gtk.Button.new_from_icon_name("document-edit-symbolic", Gtk.IconSize.BUTTON)
+        self.edit_button.set_tooltip_text(_("Edit"))
         self.edit_button.connect("clicked", self.on_edit_button_clicked)
         button_group.add_widget(self.edit_button)
         self.edit_button.set_sensitive(False)
         box.add(self.edit_button)
 
-        self.remove_button = Gtk.Button.new_with_label(_("Remove"))
+        self.remove_button = Gtk.Button.new_from_icon_name("list-remove-symbolic", Gtk.IconSize.BUTTON)
+        self.remove_button.set_tooltip_text(_("Remove"))
         self.remove_button.connect("clicked", self.on_remove_button_clicked)
         button_group.add_widget(self.remove_button)
         self.remove_button.set_sensitive(False)
         box.add(self.remove_button)
 
-        self.run_button = Gtk.Button.new_with_label(_("Run now"))
+        self.run_button = Gtk.Button.new_from_icon_name("system-run-symbolic", Gtk.IconSize.BUTTON)
+        self.run_button.set_tooltip_text(_("Run now"))
         self.run_button.connect("clicked", self.on_run_button_clicked)
         button_group.add_widget(self.run_button)
         self.run_button.set_sensitive(False)
